### PR TITLE
glib-macros: add test macro for async-glib-context tests

### PR DESCRIPTION
**Pull request by BiagioFesta**
Added a procedural macro for async tests, wrapping them in `glib::MainContext`. 
Not sure if it's useful for others, but I found myself needing this occasionally. Feel free to reject if it's not a good fit.

### CODE REVIEW

File: glib/tests/log.rs [hunk 1]
```diff
@@ -1,6 +1,10 @@
 use std::sync::{Arc, Mutex};
 
-use glib::*;
+use glib::{
+    clone, g_critical, g_debug, g_info, g_message, g_warning, log_remove_handler,
+    log_set_default_handler, log_set_fatal_mask, log_set_handler, log_unset_default_handler,
+    LogLevel, LogLevels,
+};
```
Comment (sdroege) [inline glib/tests/log.rs:7]: Why this change?

File: glib-macros/src/test.rs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    let gen_attr = quote::quote! {
+        #[::core::prelude::v1::test]
+    };
+
+    let body = &item_fn.block;
+
+    item_fn.block = syn::parse2(quote::quote! {
+        {
+            let main_ctx = glib::MainContext::default();
```
Comment (sdroege) [inline glib-macros/src/test.rs:36]: You probably want each test to run in their own main context. Otherwise running multiple tests in parallel will easily cause chaos.

File: glib-macros/src/lib.rs [hunk 3]
```diff
@@ -1567,3 +1568,22 @@ pub fn derive_value_delegate(input: TokenStream) -> TokenStream {
...
+/// An attribute macro for writing asynchronous test functions.
+///
+/// This macro is designed to wrap an asynchronous test function and ensure that
+/// it runs within a `glib::MainContext`. It helps in writing async tests that
+/// require the use of an event loop for the asynchronous execution.
+///
+/// # Example
+///
+/// ```
+/// #[glib::test]
```
Comment (sdroege) [inline glib-macros/src/lib.rs:1581]:
```suggestion
/// #[glib::async_test]
```

seems more descriptive

File: glib/tests/log.rs (hunk 1 above)
Comment (BiagioFesta) [reply glib/tests/log.rs:7]: being the macro name `test` importing `glib::*` was overriding """std""" test.

Comment (BiagioFesta) [reply glib/tests/log.rs]: anyway, renaming the macro `test -> async_test` solves this

**Review by bilelmoussaoui — APPROVED** [review]

**Review by sdroege — APPROVED** [review]
